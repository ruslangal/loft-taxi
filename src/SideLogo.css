.side-logo {
    width: 35%;

    display: flex;
    align-items: center;
    justify-content: center;

    background-color: #000;
}

.side-logo__link {
    background-image: url('./lofttaxi-biglogo.png');
    background-size: 192px 228px;
    width: 192px;
    height: 228px;
}