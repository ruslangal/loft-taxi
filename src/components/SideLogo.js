import React, { Component } from 'react'
import './SideLogo.css';
import { Link } from 'react-router-dom'

class SideLogo extends Component {
    render() {
        return (
            <div data-testid="side-logo" className="side-logo">
                <Link data-testid="side-logo__link" className="side-logo__link" to="/login"/>
            </div>
        )
    }
}

export default SideLogo;