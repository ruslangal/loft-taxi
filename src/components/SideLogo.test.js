import React from 'react';
import SideLogo from './SideLogo';
import { render, fireEvent, screen } from '@testing-library/react'

describe('SideLogo', function() {
    it('shows component with correct structure', function() {
        const MockSideLogo = () => <SideLogo navigateTo={(page) => {}} />;

        render(<MockSideLogo />);
        
        expect(screen.getByTestId('side-logo')).toMatchSnapshot();
    });

    if('calls navigatoTo function on logo click', function() {
        const navigateTo = jest.fn();
        const MockSideLogo = () => <SideLogo navigateTo={navigateTo} />;

        render(<MockSideLogo />);

        fireEvent.click(screen.getByTestId('side-logo__link'));

        expect(navigateTo).toHaveBeenCalledTimes(1);
    });
});