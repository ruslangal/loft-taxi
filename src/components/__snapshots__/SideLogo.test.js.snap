// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideLogo shows component with correct structure 1`] = `
<div
  class="side-logo"
  data-testid="side-logo"
>
  <a
    class="side-logo__link"
    data-testid="side-logo__link"
    href="#"
  />
</div>
`;
